### PR TITLE
Add MRR report

### PR DESCRIPTION
**Pull request by justnero**

### CODE REVIEW

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          $ref: ./CurrencyCode.yaml
+        breakdown:
+          description: |
+            MRR could be broken down in several categories, described below.
+            The difference between current period MRR and previous period MRR would be equal to `new + reactivated + expanded - contracted - churned`.
+          type: object
+          properties:
+            new:
+              description: |
+                MRR of new customers who didn't have revenue before they placed an order.
```
Comment (jvc9109) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:26]:
```suggestion
                MRR of new customers who didn't have recurring revenue before they placed an order.
```

Comment (adamaltman) [reply openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:26]: This seems strange. Isn't a "new customer" someone who didn't have revenue?
```suggestion
                MRR of new customers.
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+                For example, Sally is a new customer who signed up and is at $40/month plan is $40 in new MRR.
+              type: number
+              format: double
+            reactivated:
```
Comment (adamaltman) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:30]:
```suggestion
            reactivation:
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+                She purchases the $25/month plan. This is reactivation MRR, and is at $25.
+              type: number
+              format: double
+            churned:
+              description: |
+                Sally decides to cancel.
+                When the subscription churns, she went from $80/month to $0/month, and that is $80 of churned MRR.
+              type: number
+              format: double
+            contracted:
```
Comment (adamaltman) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:42]:
```suggestion
            contraction:
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+              description: |
+                Existing customers who did a change items which resulted in a smaller MRR. The contraction MRR is the difference.
+                For example, Sally was at $100 in MRR, and changed to $80 in MRR. The contraction MRR is $20.
+              type: number
+              format: double
+            expanded:
```
Comment (adamaltman) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:48]:
```suggestion
            expansion:
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+              description: |
+                Existing customers who did a change items which resulted in a bigger MRR.
+                Sally upgraded to a plan that's $100/month. She was at $40/month, so the expansions MRR is $60.
```
Comment (adamaltman) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:51]:
```suggestion
                Sally upgraded to a plan that's $100/month. She was at $40/month, so the expansion MRR is $60.
```

**Review by adamaltman — CHANGES_REQUESTED** [review]

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    type: array
+    items:
+      type: object
+      properties:
+        period:
+          description: Revenue month.
+          type: string
+          pattern: '^\d{4}-\d{2}$'
+        total:
+          description: The total MRR amount.
```
Comment (noon-dawg) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:13]: nitpick: It's good to describe the acronym in its first instance. 

```suggestion
          description: Total Maintenance Margin Requirement (MMR) amount.
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+          type: number
+          format: double
+        currency:
+          $ref: ./CurrencyCode.yaml
+        breakdown:
+          description: |
+            MRR could be broken down in several categories, described below.
+            The difference between current period MRR and previous period MRR would be equal to `new + reactivated + expanded - contracted - churned`.
```
Comment (noon-dawg) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:21]: nitpick: Avoid passives

`would, could, should, ...`

```suggestion
            MRR categories are described below.
            The difference between current period MRR and previous period MRR is equal to `new + reactivated + expanded - contracted - churned`.
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+          type: object
+          properties:
+            new:
+              description: |
+                MRR of new customers who didn't have revenue before they placed an order.
+                For example, Sally is a new customer who signed up and is at $40/month plan is $40 in new MRR.
```
Comment (noon-dawg) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:27]:
```suggestion
                For example, Sally is a new customer who signed up to a $40 per month plan. This is $40 in new MRR.
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+              type: number
+              format: double
+            reactivated:
+              description: |
+                Sally came back after previously being a customer (she churned in the past).
+                She purchases the $25/month plan. This is reactivation MRR, and is at $25.
```
Comment (noon-dawg) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:33]:
```suggestion
                Sally was previously a customer but stopped being a customer for a period.
                She signs up again and purchases a $25 per month plan. This is reactivation MRR of $25.
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+              type: number
+              format: double
+            churned:
+              description: |
+                Sally decides to cancel.
+                When the subscription churns, she went from $80/month to $0/month, and that is $80 of churned MRR.
```
Comment (noon-dawg) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:39]:
```suggestion
                A customer decides to cancel.
                The subscription churns, the customer goes from $80 per month to $0 per month. This is $80 of churned MRR.
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+              type: number
+              format: double
+            contracted:
+              description: |
+                Existing customers who did a change items which resulted in a smaller MRR. The contraction MRR is the difference.
+                For example, Sally was at $100 in MRR, and changed to $80 in MRR. The contraction MRR is $20.
```
Comment (noon-dawg) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:45]:
```suggestion
                Existing customers who changed items which resulted in a smaller MRR. Contraction MRR is the difference.
                For example, a customer at $100 in MRR changes to $80 in MRR. This contraction MRR is $20.
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml (hunk 5 above)
Comment (noon-dawg) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:51]:
```suggestion
                Existing customers who changed items which resulted in a bigger MRR.
                Example: A customer upgrades from a plan that is $40 per month to a plan that is $100 per month. This is an expansion MRR of $60.
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml (hunk 6 above)
Comment (justnero) [reply openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:13]: Agreed, but it's MRR, not MMR, so
```suggestion
          description: Total Monthly Recurring Revenue (MRR) amount.
```

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+                Example: A new customer signs up to a $40 per month plan. This is $40 in new MRR.
+              type: number
+              format: double
+            reactivation:
+              description: |
+                A previous customer stops being a customer for a period.
+                The same customer signs up again and purchases a $25 per month plan. This is reactivation MRR of $25.
+              type: number
+              format: double
+            churned:
```
Comment (justnero) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:36]: Should this also be renamed? @adamaltman @noon-dawg 
```suggestion
            churn:
```

Comment (adamaltman) [reply openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:36]: No, this is correct as churned.

**Review by adamaltman — DISMISSED** [review]
This could be made better with some examples.

File: openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml [hunk 13]
```diff
@@ -0,0 +1,53 @@
...
+properties:
+  data:
+    type: array
+    items:
+      type: object
+      properties:
+        period:
+          description: Revenue month.
+          type: string
+          pattern: '^\d{4}-\d{2}$'
```
Comment (adamaltman) [inline openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:11]: Even a simple example here would be helpful
```suggestion
          pattern: '^\d{4}-\d{2}$'
          example: 2022-06
```

Comment (justnero) [reply openapi/components/schemas/ReportMonthlyRecurringRevenue.yaml:11]: Good point, added, please re-review.

**Review by ninjasimon — APPROVED** [review]

**Review by adamaltman — APPROVED** [review]

**Review by ashkarpetin — APPROVED** [review]

**Review by noon-dawg — APPROVED** [review]

**Review by jvc9109 — APPROVED** [review]

**Pull request merged**